icx/src/main.rs

```diff
@@ -511,62 +455,77 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
             }
         }
         SubCommand::Send => {
-            let mut input: VecDeque<String> = std::io::stdin()
+            let input: VecDeque<String> = std::io::stdin()
                 .lock()
                 .lines()
                 .collect::<Result<VecDeque<String>, std::io::Error>>()?;
-            let mut line = input.pop_front().unwrap();
-            while line == "" {
-                line = input.pop_front().unwrap();
+            let mut buffer = String::new();
+            for line in input {
+                buffer.push_str(&line);
             }
 
-            let transport = ReqwestHttpReplicaV2Transport::create(opts.replica)?;
-            match line.as_str() {
-                "call" => {
-                    line = input.pop_front().unwrap(); // request id.
-                    let request_id = RequestId::from_str(&line)?;
-                    input.pop_front().unwrap(); // empty line.
-                    line = input.pop_front().unwrap(); // envelope
-                    let envelope = hex::decode(line)?;
-                    input.pop_front().unwrap(); // empty line.
-                    line = input.pop_front().unwrap(); // effective_canister_id
-                    let effective_canister_id = Principal::try_from(hex::decode(line)?)?;
-                    transport
-                        .call(effective_canister_id, envelope, request_id)
-                        .await?;
-                    eprint!("Request ID: ");
-                    println!("0x{}", hex::encode(request_id.as_slice()));
-                }
-                "read_state" => {
-                    input.pop_front().unwrap(); // empty line.
-                    line = input.pop_front().unwrap(); // envelope
-                    let envelope = hex::decode(line)?;
-                    input.pop_front().unwrap(); // empty line.
-                    line = input.pop_front().unwrap(); // effective_canister_id
-                    let effective_canister_id = Principal::try_from(hex::decode(line)?)?;
-                    let result = transport
-                        .read_state(effective_canister_id, envelope)
-                        .await?;
-                    eprint!("Result: ");
-                    println!("{}", hex::encode(result));
+            println!("{}", buffer);
+
+            let agent = Agent::builder()
+                .with_transport(
+                    agent::http_transport::ReqwestHttpReplicaV2Transport::create(
+                        opts.replica.clone(),
+                    )?,
+                )
+                .build()?;
```

Comment:
Unless I'm mistaken, here we've created two agents: one with a random key, and this one.  Could we make it so we only create the agent that we're going to use, in both cases?